apps/showcase/src/services/webcontainer/webcontainer.service.ts

```diff
@@ -126,4 +126,32 @@ export class WebContainerService {
 
     return getFilesTreeFromContainer(instance, EXCLUDED_FILES_OR_DIRECTORY);
   }
+
+  /**
+   * Retrieve all typescript declaration files from web-container
+   * @param project
+   * @param maxDepth
+   * @param path
+   */
+  public async getDeclarationTypes(project: string, maxDepth = 20, path = 'node_modules'): Promise<{ filePath: string; content: string }[]> {
+    // TODO read that from project devDependencies?
+    const dependenciesWhiteList = /@ama-sdk|@angular|@o3r|rxjs|@types/;
```

Comment:
why not using array instead of regexp?

Reply:
I though it might be more convenient if we want to use patterns on the files like `@angular.*` or something

Reply:
I think an array + startsWith would be easier to maintain.
(btw, if you keep regex you may want to add `^` because if it does not contains `@` it may match unwanted deps)

Reply:
I've added the `^` and `$`
anyway, this may change after we implement the TODO about dev deps